### PR TITLE
[feat] Add global loading indicators

### LINKED ISSUE #124: loading indicators
https://linear.app/trap-garden/issue/TG-78/add-loading-indicators-globally

### PR DESCRIPTION
**Pull request by SashankBalusu**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
this was so odd

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @kylezryr

### CODE REVIEW

File: components/CircularLoader/loader.tsx
Comment (kylezryr) [inline components/CircularLoader/loader.tsx:1]: is there a reason this file is called loader.tsx instead of index? does it not work if its called index

File: app/view-plants/page.tsx [hunk 1]
```diff
@@ -183,12 +187,16 @@ export default function Page() {
...
+      startTransition(() => {
+        router.push(`${CONFIG.generalPlant}/${plant.id}`);
+      });
     }
   }
   function handleAddPlants() {
     setPlantsToAdd(selectedPlants);
-    router.push('/add-details');
+    startTransition(() => {
+      router.push('/add-details');
```
Comment (kylezryr) [inline app/view-plants/page.tsx]: add this route to configs and use `CONFIG.addDetails` instead

File: app/(auth)/signup/page.tsx [hunk 2]
```diff
@@ -200,9 +208,9 @@ export default function SignUp() {
...
               onClick={handleSignUp}
               $primaryColor={COLORS.shrub}
-              disabled={!canSubmitForm}
+              disabled={!canSubmitForm || isSignupButtonDisabled}
             >
-              <BigButtonText $color="white">Sign Up</BigButtonText>
+              <BigButtonText $color="white">{signupButtonText}</BigButtonText>
             </BigButton>
           </div>
         </StyledForm>
```
Comment (kylezryr) [inline app/(auth)/signup/page.tsx:226]: all auth loading looks good

File: components/RandomVeggieLoader/loader.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    transform: translateX(-50%);
+    border-left: 10px solid transparent;
+    border-right: 10px solid transparent;
+    border-bottom: 22px solid #3fa34d;
+    filter: blur(0.3px);
+  }
+`;
+
+/* ═══════════════════════ 3. CAULIFLOWER ═══════════════════════ */
+/* — Cauliflower — */
```
Comment (kylezryr) [inline components/RandomVeggieLoader/loader.tsx]: i lowk dont fuck with the cauliflower... maybe remove this one big dawg

**Review by kylezryr — CHANGES_REQUESTED** [review]
works well, dw too much about edge cases. the nav column closing too early is def a noticeable problem tho, could we look into that? more specifically, the nav column closes after clicking an item but no loading indication is shown for a couple of seconds. good job otherwise

File: components/RandomVeggieLoader/loader.tsx (hunk 3 above)
Comment (SashankBalusu) [reply components/RandomVeggieLoader/loader.tsx]: yes master
